### PR TITLE
A new process for mapping usernames to openids

### LINKED ISSUE #589: myproxy_logon generate certificates without populating the CN
The CN should be the complete OpenID, `https://{openid-sever}/esgf-idp/openid/{username}`. This is found by querying the database with a Java application, which is called from a script at `/esg/config/myproxy/myproxy-certificate-mapapp`. Manually running this script from `/usr/local/globus/bin` reports the following error:
```
An error occurred: java.lang.Exception: Openid Query Failed: org.postgresql.util.PSQLException: ERROR: permission denied for schema esgf_security
  Position: 29
```
It was discovered that the database user being used is the low-privilege user, where as in 2.x, the database user being used is the database super user. It is supposed to be referencing the `db.user` property in the `/esg/config/esgf.properties` file and uses the following command.
```
postgress_user=$(sed -n 's@^[^#]*[ ]*db.user[ ]*=[ ]*\(.*\)$@\1@p' ${ESGF_HOME}/config/esgf.properties | head -n1)
```
Where `ESGF_HOME`  is `/esg`. This resolves to the low privilege user because `/esg/config/esgf.properties` contains
```
...
publisher.db.user = {low privilege user}
...
db.user = {the super db user}
...
```

Note this is related #482 which proposes to change the `/esg/config/myproxy/myproxy-certificate-mapapp` to a standalone python script which makes the query itself.

### PR DESCRIPTION
**Pull request by nathanlcarlson**
This is to replace the old java application that performed this process and will resolve #589 and #482. It still needs to be copied into place, `/esg/config/myproxy/`, by the installer.

### CODE REVIEW

**Comment by nathanlcarlson** [comment]
While the scripts themselves have been manually tested, this should be run with a full install.

File: idp_node/mapapp/openid_retriever.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    section = "installer.properties"
+
+    db = config_parser.get(section, "db.database")
+    host = config_parser.get(section, "db.host")
+    port = config_parser.get(section, "db.port")
+    user = config_parser.get(section, "db.user")
+    esgf_host = config_parser.get(section, "esgf.host")
+    gateway = "https://{}/esgf-idp/openid/%".format(esgf_host)
+
+    cs = "dbname=%s user=%s password=%s host=%s port=%s" % (db, user, password, host, port)
```
Comment (William-Hill) [inline idp_node/mapapp/openid_retriever.py]: Variable name should be more descriptive for readability

File: idp_node/mapapp/openid_retriever.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    cs_no_pass = "dbname=%s user=%s host=%s port=%s" % (db, user, host, port)
+    logging.info(cs_no_pass)
```
Comment (William-Hill) [inline idp_node/mapapp/openid_retriever.py]: This logging message could have additional context.  Something like "fetching OpenID for {some_user}"

File: idp_node/mapapp/openid_retriever.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    cert = OpenSSL.crypto.load_certificate(OpenSSL.crypto.FILETYPE_PEM, cert_contents)
+    return cert.get_subject()
+
+def main():
+
+    parser = argparse.ArgumentParser(
+        description="A script for querying the esgcet database for an OpenID, given a username"
+    )
+    parser.add_argument("username",  help="The username for which to query for the OpenID")
+    parser.add_argument("cert_file",  help="The username for which to query for the OpenID")
```
Comment (William-Hill) [inline idp_node/mapapp/openid_retriever.py]: Help statement should be updated. Looks like a copy/paste error

**Pull request merged**